components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/factory/LogoutResponseFactory.java

```diff
@@ -0,0 +1,121 @@
+import org.wso2.carbon.identity.application.authentication.framework.inbound.IdentityResponse;
+import org.wso2.carbon.identity.application.authenticator.oidc.logout.idpinit.exception.LogoutClientException;
+import org.wso2.carbon.identity.application.authenticator.oidc.logout.idpinit.exception.LogoutException;
+import org.wso2.carbon.identity.application.authenticator.oidc.model.LogoutResponse;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+import static org.wso2.carbon.identity.application.authenticator.oidc.util.OIDCErrorConstants.ErrorMessages.LOGOUT_CLIENT_EXCEPTION;
+import static org.wso2.carbon.identity.application.authenticator.oidc.util.OIDCErrorConstants.ErrorMessages.LOGOUT_SERVER_EXCEPTION;
+
+/**
+ * Builds a HTTP response instance based on the common IdentityRequest format used by
+ * the authentication framework.
+ */
+public class LogoutResponseFactory extends HttpIdentityResponseFactory {
+
+    private static final Log log = LogFactory.getLog(LogoutResponseFactory.class);
+
+    @Override
+    public boolean canHandle(IdentityResponse identityResponse) {
+
+        return (identityResponse instanceof LogoutResponse);
+
+    }
+
+    @Override
+    public boolean canHandle(FrameworkException exception) {
+
+        return (exception instanceof LogoutException);
+    }
+
+    @Override
+    public HttpIdentityResponse.HttpIdentityResponseBuilder create(IdentityResponse identityResponse) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder httpIdentityResponseBuilder = new HttpIdentityResponse
+                .HttpIdentityResponseBuilder();
+        create(httpIdentityResponseBuilder, identityResponse);
+        return httpIdentityResponseBuilder;
+    }
+
+    @Override
+    public void create(HttpIdentityResponse.HttpIdentityResponseBuilder builder, IdentityResponse identityResponse) {
+
+        builder.setStatusCode(((LogoutResponse) identityResponse).getStatusCode());
+        builder.addHeader(OAuthConstants.HTTP_RESP_HEADER_CACHE_CONTROL,
+                OAuthConstants.HTTP_RESP_HEADER_VAL_CACHE_CONTROL_NO_STORE);
+        builder.addHeader(OAuthConstants.HTTP_RESP_HEADER_PRAGMA,
+                OAuthConstants.HTTP_RESP_HEADER_VAL_PRAGMA_NO_CACHE);
+        builder.addHeader(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_PLAIN);
+        builder.setBody(((LogoutResponse) identityResponse).getMessage());
+    }
+
+    @Override
+    public HttpIdentityResponse.HttpIdentityResponseBuilder handleException(FrameworkException frameworkException) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder builder;
+        if (frameworkException instanceof LogoutClientException) {
+            if (log.isDebugEnabled()) {
+                log.debug(LOGOUT_CLIENT_EXCEPTION.getMessage() + ":" + frameworkException.getMessage(),
```

Comment:
```suggestion
                log.debug(LOGOUT_CLIENT_EXCEPTION.getMessage() + ": " + frameworkException.getMessage(),
```

Reply:
To increase readability i think we can go with something like

"Client error when handling the request: " + frameworkException.getMessage()